chore(OTEL): move troubleshooting topics to separate pages (#17055)
newrelic/docs-website: src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/troubleshooting/missing-entities-relationships.mdx

---
title: "Troubleshoot OpenTelemetry: Missing entities or relationships"
tags:
  - Integrations
  - Open source telemetry integrations
  - OpenTelemetry
  - Troubleshooting
metaDescription: Troubleshoot tips for when you use OpenTelemetry with New Relic
freshnessValidatedDate: 2024-05-01
---

## Problem

You sent OpenTelemetry data from a service or infrastructure component, and either the [entity](/docs/new-relic-one/use-new-relic-one/core-concepts/what-entity-new-relic/) or its [relationships](/docs/new-relic-solutions/new-relic-one/core-concepts/what-entity-new-relic/#related-entities) are missing or incorrect in the UI.

## Solution

> OpenTelemetry entities are synthesized based on the public rules described for the [`EXT-SERVICE`](https://github.com/newrelic/entity-definitions/blob/b8e75d839eed7859044a9bb601a62e5323107350/definitions/ext-service/definition.yml) entity type. The standard rule to match relies on the presence of the `service.name` dimension (which follows the OpenTelemetry [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/527206ea49e384de957eda105d9425aa8fefca25/specification/overview.md#semantic-conventions)).

Author: "The standard rule to match relies on the presence of the `service.name` dimension (which follows the OpenTelemetry [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/527206ea49e384de957eda105d9425aa8fefca25/specification/overview.md#semantic-conventions))" is a bit too passive. How about this revision for clarity?

```suggestion
OpenTelemetry entities are synthesized based on the public rules described for the [`EXT-SERVICE`](https://github.com/newrelic/entity-definitions/blob/b8e75d839eed7859044a9bb601a62e5323107350/definitions/ext-service/definition.yml) entity type. How the entity name appears in New Relic is based on the `service.name` dimension, which follows the OpenTelemetry [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/527206ea49e384de957eda105d9425aa8fefca25/specification/overview.md#semantic-conventions)).
```

> To set the `service.name` with the OpenTelemetry Java SDK, include it in your [resource](/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-concepts/#resources):

Author: Is this available with other language SDKs, or just Java?

---

> title: "Troubleshoot OpenTelemetry: Missing entities or relationships"

Reviewer: This document is limited to 1) limited to service entities and 2) barely addresses relationships in any meaningful way.

A document about entities and relationships is sorely needed. Though, it's strange as a "troubleshooting" doc. We really need a top-level document addressing this subject holistically. Including, service, host, and other entity types that can be synthesized from otel data. Plus how relationships work between these entity types.

metaDescription: Troubleshoot tips for when you use OpenTelemetry with New Relic
freshnessValidatedDate: 2024-05-01
---

## Problem

You sent OpenTelemetry data from a service or infrastructure component, and either the [entity](/docs/new-relic-one/use-new-relic-one/core-concepts/what-entity-new-relic/) or its [relationships](/docs/new-relic-solutions/new-relic-one/core-concepts/what-entity-new-relic/#related-entities) are missing or incorrect in the UI.

## Solution

OpenTelemetry entities are synthesized based on the public rules described for the [`EXT-SERVICE`](https://github.com/newrelic/entity-definitions/blob/b8e75d839eed7859044a9bb601a62e5323107350/definitions/ext-service/definition.yml) entity type. The standard rule to match relies on the presence of the `service.name` dimension (which follows the OpenTelemetry [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/527206ea49e384de957eda105d9425aa8fefca25/specification/overview.md#semantic-conventions)).

* For Java: Include the  `service.name` in your resource:

```java
var resource = Resource.getDefault()
    .merge(Resource.builder().put(SERVICE_NAME, serviceName).build());
```

>     <Callout variant="tip" title="About resources">

Author: ```suggestion
    <Callout variant="tip">
```

        The following suites of attributes are defined by the [OpenTelemetry resource semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/527206ea49e384de957eda105d9425aa8fefca25/specification/overview.md#semantic-conventions). These attributes are usually set by creating a resource using the OpenTelemetry SDK.

            * `service.*` attributes
                * `service.name` attribute is required to associate your resource with an entity in the UI
                * `service.instance.id` is required for certain panes to light up
            * `telemetry.sdk.language=java` is required to see data in the JVM section
        </Callout>

* For all other SDK languages, set `service.name` by declaring it in the `OTEL_RESOURCE_ATTRIBUTES` or `OTEL_SERVICE_NAME` [environment variables](https://github.com/open-telemetry/opentelemetry-specification/blob/20c82de552d08428e8cadaaef3e6cb46812f7c00/specification/sdk-environment-variables.md#general-sdk-configuration).

> For New Relic <InlinePopover type="logs" />, you can use a structured log template to inject the `service.name`. See [Logs in context with Log4j2](https://github.com/newrelic/newrelic-opentelemetry-examples/blob/e3f5ee85b4dcd8dd29f8f69d78d122b82a9638ba/other-examples/java/logs-in-context-log4j2/Log4j2EventLayout.json#L2) for an example.

Author: When would they inject this via logs vs. declaring it in the `OTEL_RESOURCE_ATTRIBUTES` or `OTEL_SERVICE_NAME?

<Callout variant="tip">

>   For more OpenTelemetry examples with New Relic, visit the [newrelic-opentelemetry-examples](https://github.com/newrelic/newrelic-opentelemetry-examples) repository on GitHub.

Author: This needs a better home/context (maybe after the "For all other SDKs...)
